pkg/controllers/pod_controller.go

```diff
@@ -98,6 +98,15 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	// validate leader annotations to prevent infinite StatefulSet creation loops
+	// see issue: https://github.com/kubernetes-sigs/lws/issues/391
+	if pod.Annotations[leaderworkerset.LeaderPodNameAnnotationKey] != "" {
```

Comment:
I think we should log an error, emit an event with the error. 

However, returning an error is not useful here since it will put the leader pod back into the workerqueue, which means it will get reconciled again and again forever, so I think we should not return an error here and in the error message we log, we should say that we are giving up on reconciling the pod.

Reply:
Thanks @ahg-g 

It has been changed :-)